pages/deployments/deploying_to_kubernetes.md.erb

```diff
@@ -0,0 +1,222 @@
+# Deploying to Kubernetes
+
+<%= estimated_time "30 minutes" %>
+
+<%= toc %>
+
+This tutorial shows how to deploy to Kubernetes using separate test and deploy
+[pipelines][]. Separating pipelines like this is best practice. The test
+pipeline runs tests and push a Docker image to a registry. The deploy pipelines
+uses the `DOCKER_IMAGE` enviornment variable to create a [Kubernetes
+deployment][k8s_deployment] via `kubectl`. Then, you'll see how to link them
+together to automate deploys from master.
+
+First up, you need to add a step to your existing test pipline that pushes a
+Docker image. Also check your agents have `kubectl` access to your target
+cluster. Refer to the notes at the end of tutorial for tips on setting this
+up.
+
+<%= image "final_test_pipeline.png", width: 736, height: 199, alt: 'Final Test Pipeline' %>
+
+Let's go.
+
+## Create the Deploy Pipeline
+
+This section covers creating a new Buildkite pipeline that loads steps from
+`.buildkite/pipeline.deploy.yml`. Deploy pipelines are like any other pipeline
+except they're triggred by events other than code pushes. We'll use a [trigger
+steps](https://buildkite.com/docs/pipelines/trigger-step) later on to connect
+the test and deploy pipelines.
+
+Create a [new pipeline](https://buildkite.com/new) for your repo. Use
+`buildkite-agent pipeline upload .buildkite/pipeline.deploy.yml` for the first
+step as shown in the image.
+
+<%= image "new_pipeline.png", width: 939/2, height: 454/2, alt: 'Creating a New Pipelien'  %>
+
+Now create `.buildkite/pipeline.deploy.yml` with a single step. We'll write the
+deploy script in the next step.
+
+```yml
+steps:
+  - label: "\:rocket\: Push to \:kubernetes\:"
+    comamnd: script/buildlkite/deploy
+    concurrency: 1
+    concurrency_group: deploy/tutorial
+```
+
+Set `concurrency` and `concurrency_group` when updating mutable state. These
+settings ensures only one step runs at a time, effectively creating a deploy
```

Comment:
'Queue' has a specific meaning in buildkite, how about we just end that sentence at 'time'?